autospec/files.py

```diff
@@ -184,144 +317,24 @@ def push_file(self, filename):
-            (r"^/usr/lib64/[a-zA-Z0-9\.\_\-\+]*\.a$", "dev", "/usr/lib64/*.a"),
-            (r"^/usr/lib32/[a-zA-Z0-9\.\_\-\+]*\.a$", "dev32", "/usr/lib32/*.a"),
-            (r"^/usr/lib/haswell/[a-zA-Z0-9\.\_\-\+]*\.a$", "dev", "/usr/lib/haswell/*.a"),
-            (r"^/usr/lib64/haswell/[a-zA-Z0-9\.\_\-\+]*\.a$", "dev", "/usr/lib64/haswell/*.a"),
-            (r"^/usr/lib32/haswell/[a-zA-Z0-9\.\_\-\+]*\.a$", "dev32", "/usr/lib32/haswell/*.a"),
-            (r"^/usr/lib/pkgconfig/[a-zA-Z0-9\.\_\-\+]*\.pc$", "dev"),
-            (r"^/usr/lib64/pkgconfig/[a-zA-Z0-9\.\_\-\+]*\.pc$", "dev"),
-            (r"^/usr/lib32/pkgconfig/[a-zA-Z0-9\.\_\-\+]*\.pc$", "dev32"),
-            (r"^/usr/lib/[a-zA-Z0-9\.\_\-\+]*\.la$", "dev"),
-            (r"^/usr/lib64/[a-zA-Z0-9\.\_\-\+]*\.la$", "dev"),
-            (r"^/usr/lib32/[a-zA-Z0-9\.\_\-\+]*\.la$", "dev32"),
-            (r"^/usr/lib/[a-zA-Z0-9\.\_\-\+]*\.prl$", "dev"),
-            (r"^/usr/lib64/[a-zA-Z0-9\.\_\-\+]*\.prl$", "dev"),
-            (r"^/usr/lib32/[a-zA-Z0-9\.\_\-\+]*\.prl$", "dev32"),
-            (r"^/usr/share/aclocal/[a-zA-Z0-9\.\_\-\+]*\.ac$", "dev", "/usr/share/aclocal/*.ac"),
-            (r"^/usr/share/aclocal/[a-zA-Z0-9\.\_\-\+]*\.m4$", "dev", "/usr/share/aclocal/*.m4"),
-            (r"^/usr/share/aclocal-1.[0-9]+/[a-zA-Z0-9\.\_\-\+]*\.ac$", "dev", "/usr/share/aclocal-1.*/*.ac"),
-            (r"^/usr/share/aclocal-1.[0-9]+/[a-zA-Z0-9\.\_\-\+]*\.m4$", "dev", "/usr/share/aclocal-1.*/*.m4"),
-            (r"^/usr/share/doc/" + re.escape(tarball.name) + "/", "doc", "%doc /usr/share/doc/" + re.escape(tarball.name) + "/*"),
-            (r"^/usr/share/doc/", "doc"),
-            (r"^/usr/share/gtk-doc/html", "doc"),
-            (r"^/usr/share/help", "doc"),
-            (r"^/usr/share/info/", "doc", "%doc /usr/share/info/*"),
-            (r"^/etc/systemd/system/.*\.wants/", "active-units"),
-            # now a set of catch-all rules
-            (r"^/etc/", "config", "", "%config "),
-            (r"^/usr/etc/", "config", "", "%config "),
-            (r"^/lib/systemd", "config"),
-            (r"^/usr/lib/systemd", "config"),
-            (r"^/usr/lib/udev/rules.d", "config"),
-            (r"^/usr/lib/modules-load.d", "config"),
-            (r"^/usr/lib/tmpfiles.d", "config"),
-            (r"^/usr/lib/sysusers.d", "config"),
-            (r"^/usr/lib/sysctl.d", "config"),
-            (r"^/usr/share/", "data"),
-            # finally move any dynamically loadable plugins (not
-            # perl/python/ruby/etc.. extensions) into lib package
-            (r"^/usr/lib/.*/[a-zA-Z0-9\.\_\-\+]*\.so", "lib"),
-            (r"^/usr/lib64/.*/[a-zA-Z0-9\.\_\-\+]*\.so", "lib"),
-            (r"^/usr/lib32/.*/[a-zA-Z0-9\.\_\-\+]*\.so", "lib32"),
-            # locale data gets picked up via file_is_locale
-            (r"^/usr/share/locale/", "ignore")]
+        if self.want_dev_split and self.file_pat_match(filename, re.compile(r"^/usr/.*/include/.*\.h$"), "dev"):
```

Comment:
Since we have a nice little object we could initialize these static regex strings there, not required for this commit though but maybe a todo for later.